### PR TITLE
SP ARM32: support for arch with no ldrd/strd and clz

### PR DESCRIPTION
**Pull request by SparkiDev**
# Description

Defined WOLFSSL_SP_ARM_ARCH to be less than 7 to use alternative
instructions.

# Testing

./configure --disable-shared --enable-sp=yes,asm 'CC=arm-linux-gnueabi-gcc' -host=armv8 LDFLAGS=--static "C_EXTRA_FLAGS=-DWOLFSSL_SP_ARM_ARCH=6"

# Checklist

 - [ ] added tests
 - [ ] updated/added doxygen
 - [ ] updated appropriate READMEs
 - [ ] Updated manual and documentation

### CODE REVIEW

File: wolfcrypt/src/sp_x86_64_asm.S [hunk 1]
```diff
@@ -20,7 +20,7 @@
  */
 
 #ifdef WOLFSSL_USER_SETTINGS
-#include "wolfssl/wolfcrypt/settings.h"
+#include "wolfssl/wolfcrpyt/settings.h"
```
Comment (dgarske) [inline wolfcrypt/src/sp_x86_64_asm.S:23]: Not sure what happened here... `wolfcrypt` -> `wolfcrpyt`? Guessing this line needs reverted.

Comment (dgarske) [reply wolfcrypt/src/sp_x86_64_asm.S:23]: Fixed here: https://github.com/wolfSSL/scripts/pull/245

**Review by dgarske — APPROVED** [review]

File: wolfcrypt/src/sp_x86_64_asm.S (hunk 1 above)
Comment (SparkiDev) [reply wolfcrypt/src/sp_x86_64_asm.S:23]: Changed script but code got checked in by someone else.
Fixed now.

**Review by dgarske — APPROVED** [review]
Very nice fixes adding `WOLFSSL_SP_ARM_ARCH`. Built and tested. Thanks @SparkiDev

**Pull request merged**